MPF.Core/Modules/DiscImageCreator/Parameters.cs

```diff
@@ -548,6 +548,8 @@ public override void GenerateSubmissionInfo(SubmissionInfo info, Options options
                         info.CommonDiscInfo.Region = info.CommonDiscInfo.Region ?? pythonTwoRegion;
                         info.CommonDiscInfo.EXEDateBuildDate = pythonTwoDate;
                     }
+                    if (info.CommonDiscInfo!.CommentsSpecialFields!.TryGetValue(SiteCode.InternalSerialName, out string? kp2Serial))
```

Comment:
Given that you need to do modifications on the serial at each point to get the date, it seems reasonable to split the code that gets the executable name from both the part that gets the serial and the part that gets the date. This will make it so that you don't have to worry about overwriting with the wrong value or in the wrong order. This goes for all 3 in DIC.

Reply:
Split GetExecutableName from GetExecutableInfo